### PR TITLE
Adding sonarqube to shared services

### PR DESCRIPTION
**Pull request by steeeveo**
* Adds sonarqube deployment to shared services cluster
* Handles some errors from the latest AMI version used by the cluster
* Adds an additional external-dns release to handle the public zone
* Adds external-facing ingress components
* Upgrades the external-dns release to the latest version

### CODE REVIEW

File: stages/apps/sharedsvc/sonarqube.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+data "vault_generic_secret" "sonarqube" {
+  path = "lead/aws/265560927720/sonarqube"
```
Comment (mrparkers) [inline stages/apps/sharedsvc/sonarqube.tf:2]: we should derive the account number from the [`caller_identity` data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/caller_identity)

File: modules/environment/aws/eks/main.tf [hunk 2]
```diff
@@ -9,7 +9,7 @@ systemctl enable amazon-ssm-agent
 
 echo '{"registry-mirrors": [${ var.docker_registry_mirror != "" ? format("\"%s\"", var.docker_registry_mirror) : ""}]}' | cat /etc/docker/daemon.json - | jq -s '.[0] * .[1]' > /tmp/daemon.json
 mv /tmp/daemon.json /etc/docker/daemon.json
-systemctl reload docker
+systemctl restart docker
```
Comment (mrparkers) [inline modules/environment/aws/eks/main.tf:12]: let's revert this change, we want this to be changed in the upcoming PR to upgrade the LEAD clusters to 1.17

**Review by mrparkers — CHANGES_REQUESTED** [review]

File: stages/apps/sharedsvc/sonarqube.tf (hunk 1 above)
Comment (steeeveo) [reply stages/apps/sharedsvc/sonarqube.tf:2]: The hardcoding was intentional; as we didn't see a reason to make it dynamic, but I'm fine making it dynamic.

**Review by mrparkers — APPROVED** [review]
